modules/lib/src/main/resources/assets/js/app/view/ContentItemPreviewPanel.ts

```diff
@@ -76,19 +77,19 @@ export class ContentItemPreviewPanel
         return item as ContentSummaryAndCompareStatus;
     }
 
-    protected doSetItem(item: ViewItem, force: boolean = false) {
+    protected async doSetItem(item: ViewItem, force: boolean = false) {
         const content = this.viewItemToContent(item);
 
-        if (this.isPreviewUpdateNeeded(content, force)) {
+        if (await this.isPreviewUpdateNeeded(content, force)) {
             this.update(content);
         }
 
         this.toolbar.setItem(item);
         this.item = item;
     }
 
-    public isPreviewUpdateNeeded(item: ContentSummaryAndCompareStatus, force?: boolean): boolean {
-        return !this.skipNextSetItemCall && this.isItemAllowsUpdate(item, force);
+    public async isPreviewUpdateNeeded(item: ContentSummaryAndCompareStatus, force?: boolean): Promise<boolean> {
+        return !this.skipNextSetItemCall && (this.isItemAllowsUpdate(item, force) || this.isRenderable());
```

Comment:
Values of different types should not be combined in one logical expression. The method used to return `boolean && boolean`, now you changed it to return `Promise<boolean>` but the condition inside is `boolean && (boolean ||  Promise<boolean>)`.

First of all, Promise should be replaced with Q.Promise as we are not on native promises yet.
Then the condition should be split in several so that the method always returns a promise, not a boolean.

Reply:
Logical operations work correctly with native promises returning a promise if one of the operands is one.

But I am not sure about Q.Promise, so will change it since we're going back to Q.

Reply:
Which I am not a fan of btw